### DISCUSSION TITLE
Hot reload with Json and getStaticProps

### DISCUSSION THREAD

**Discussion by mquandalle** [vercel/next.js#12149] [Help]
Hello,

My `getStaticProps` function reads a json file on the file system. When this file is modified I would like the client side to hot reload with the new data.

Currently the file modification triggers a compilation (with the message `[ ready ] compiled successfully` in the console) but the web client doesn't hot reload, I have to refresh the page.

Any pointer?

**Comment by yantakus** [comment]
Hi @mquandalle. Were you able to solve this problem? Issuing it as well.

**Reply by mquandalle** [reply]
No, I wasn't able to find a warkaround

**Comment by yantakus** [comment]
I can add that this happens not only with .json files. If we import something from .js/.ts files (for example translations) inside of `getStaticProps`, hot reload doesn't happen as well. Need help.

**Comment by spaceemotion** [comment]
Just had the same issue with a dynamically generated .json file by another process in the background. Would love to see hot reloads (or at least automatic page reloads).

**Comment by henryhobhouse** [comment]
Also in the same boat with markdown in that no hot reload when the file/s that are retrieved in getStatisProps are updated. Would be great if we could extend the api of which files are monitored for change to trigger a hot reload.

**Comment by kripod** [comment]
I just came across this issue while browsing for hot reloading markdown files, just like @henryhobhouse. Has anyone been able to find a solution for this issue since it has been posted originally?

**Comment by bluprince13** [comment]
I used https://github.com/hashicorp/next-remote-watch to solve this for my blog where I wanted hot reloading for my markdown files.

My commit: https://github.com/bluprince13/bluprince13-next/commit/941c06ecdfce85e47b09a728ae89443a7a753619

**Reply by HerbCaudill** [reply]
+1 for next-remote-watch - works like a charm

**Comment by gerardvanderput** [comment]
This solution triggers a hot-reload when your JSON file changes:
https://gist.github.com/gerardvanderput/88b669cdcac302b8ab928754edcb75d2

TLDR:
- add a **useEffect** hook to your component
- which imports the JSON file dynamically (see below), but only when NODE_ENV eq. `development`.

```
await import("/data/my/file.json");
```

Note: I don't use the result that's returned by this call. That's done elsewhere (and passed to getStaticProps). But adding this hook, will trigger the reload.
